File: mobiledevelopment/lib/src/config/roles.dart
import 'package:flutter/material.dart';
import '../styles/theme.dart';
import '../types/index.dart';

export '../types/index.dart' show UserRoles;

const roleLabels = {
  UserRoles.superAdmin: 'Super Admin',
  UserRoles.marketingManager: 'Marketing Manager',
  UserRoles.contentCreator: 'Content Creator',
  UserRoles.marketingAnalyst: 'Marketing Analyst',
};

/// Roles users may self-select at registration — Super Admin is admin-assigned only.
const registrationRoles = [
  UserRoles.marketingManager,
  UserRoles.contentCreator,
  UserRoles.marketingAnalyst,
];

Map<String, String> get registrationRoleLabels => Map.fromEntries(
  registrationRoles.map((role) => MapEntry(role, roleLabels[role]!)),
);

bool isRegistrationRole(String role) => registrationRoles.contains(role);

const roleDashboardPaths = {
  UserRoles.superAdmin: '/super-admin',
  UserRoles.marketingManager: '/marketing-manager',
  UserRoles.contentCreator: '/content-creator',
  UserRoles.marketingAnalyst: '/marketing-analyst',
};

String getDashboardPath(String role) => roleDashboardPaths[role] ?? '/login';
String getRoleLabel(String role) => roleLabels[role] ?? role;

class RoleBadgeColors {
  final Color background;
  final Color foreground;
  final Color? border;
  const RoleBadgeColors({required this.background, required this.foreground, this.border});
}

const roleColors = {
  UserRoles.superAdmin: RoleBadgeColors(background: Color(0x1A164478), foreground: AppColors.brand900, border: AppColors.brand200),
  UserRoles.marketingManager: RoleBadgeColors(background: AppColors.brand100, foreground: AppColors.brand800),
  UserRoles.contentCreator: RoleBadgeColors(background: AppColors.diamond100, foreground: AppColors.diamond600),
  UserRoles.marketingAnalyst: RoleBadgeColors(background: AppColors.brand50, foreground: AppColors.brand700, border: AppColors.brand200),
};

Widget roleBadge(String role) {
  final colors = roleColors[role] ?? RoleBadgeColors(background: const Color(0xFFF1F5F9), foreground: AppColors.slate700);
  return Container(
    padding: const EdgeInsets.symmetric(horizontal: 10, vertical: 5),
    decoration: BoxDecoration(
      color: colors.background,
      borderRadius: BorderRadius.circular(999),
      border: colors.border != null ? Border.all(color: colors.border!) : Border.all(color: colors.foreground.withValues(alpha: 0.12)),
    ),
    child: Text(getRoleLabel(role), style: TextStyle(color: colors.foreground, fontSize: 11, fontWeight: FontWeight.w800, letterSpacing: 0.2)),
  );
}

class RoleQuickAccessStyle {
  final Color cardBorder;
  final Color cardBackground;
  final Gradient iconGradient;
  final Color buttonBackground;
  final Color buttonForeground;
  const RoleQuickAccessStyle({
    required this.cardBorder,
    required this.cardBackground,
    required this.iconGradient,
    required this.buttonBackground,
    required this.buttonForeground,
  });
}

const roleQuickAccess = {
  UserRoles.superAdmin: RoleQuickAccessStyle(
    cardBorder: Color(0xCC93C5FD),
    cardBackground: Color(0xB3EEF4FC),
    iconGradient: LinearGradient(colors: [AppColors.brand800, AppColors.brand900]),
    buttonBackground: AppColors.brand900,
    buttonForeground: Colors.white,
  ),
  UserRoles.marketingManager: RoleQuickAccessStyle(
    cardBorder: Color(0xCCBFDBFE),
    cardBackground: Color(0x99EEF4FC),
    iconGradient: LinearGradient(colors: [AppColors.brand500, AppColors.brand700]),
    buttonBackground: AppColors.brand600,
    buttonForeground: Colors.white,
  ),
  UserRoles.contentCreator: RoleQuickAccessStyle(
    cardBorder: Color(0xCCB8E4F2),
    cardBackground: Color(0x99E0F4FA),
    iconGradient: LinearGradient(colors: [AppColors.diamond400, AppColors.brand500]),
    buttonBackground: AppColors.diamond500,
    buttonForeground: Colors.white,
  ),
  UserRoles.marketingAnalyst: RoleQuickAccessStyle(
    cardBorder: Color(0xCCB8E4F2),
    cardBackground: Color(0x80E0F4FA),
    iconGradient: LinearGradient(colors: [AppColors.diamond200, AppColors.diamond500]),
    buttonBackground: AppColors.diamond600,
    buttonForeground: Colors.white,
  ),
};

class RoleSidebarAccent {
  final Gradient activeGradient;
  final Color iconColor;
  const RoleSidebarAccent({required this.activeGradient, required this.iconColor});
}

const roleSidebarAccent = {
  UserRoles.superAdmin: RoleSidebarAccent(
    activeGradient: LinearGradient(colors: [AppColors.brand800, AppColors.brand900]),
    iconColor: AppColors.brand800,
  ),
  UserRoles.marketingManager: RoleSidebarAccent(
    activeGradient: LinearGradient(colors: [AppColors.brand600, AppColors.brand800]),
    iconColor: AppColors.brand600,
  ),
  UserRoles.contentCreator: RoleSidebarAccent(
    activeGradient: LinearGradient(colors: [AppColors.diamond500, AppColors.brand600]),
    iconColor: AppColors.diamond600,
  ),
  UserRoles.marketingAnalyst: RoleSidebarAccent(
    activeGradient: LinearGradient(colors: [AppColors.diamond400, AppColors.brand500]),
    iconColor: AppColors.diamond600,
  ),
};

const roleActionAccent = {
  UserRoles.superAdmin: LinearGradient(colors: [AppColors.brand100, AppColors.brand200]),
  UserRoles.marketingManager: LinearGradient(colors: [AppColors.brand100, Color(0xFF9BBFE8)]),
  UserRoles.contentCreator: LinearGradient(colors: [AppColors.diamond100, AppColors.diamond200]),
  UserRoles.marketingAnalyst: LinearGradient(colors: [AppColors.diamond100, AppColors.brand200]),
};

File: src/config/roles.ts
import type { UserRole } from '../types';

export const ROLE_LABELS: Record<UserRole, string> = {
  'super-admin': 'Super Admin',
  'marketing-manager': 'Marketing Manager',
  'content-creator': 'Content Creator',
  'marketing-analyst': 'Marketing Analyst',
};

/** Roles users may self-select at registration — Super Admin is admin-assigned only. */
export const REGISTRATION_ROLES: UserRole[] = [
  'marketing-manager',
  'content-creator',
  'marketing-analyst',
];

export function getRegistrationRoleLabels(): Record<UserRole, string> {
  return Object.fromEntries(REGISTRATION_ROLES.map((role) => [role, ROLE_LABELS[role]])) as Record<UserRole, string>;
}

export function isRegistrationRole(role: string): role is UserRole {
  return REGISTRATION_ROLES.includes(role as UserRole);
}

export const ROLE_DASHBOARD_PATHS: Record<UserRole, string> = {
  'super-admin': '/super-admin',
  'marketing-manager': '/marketing-manager',
  'content-creator': '/content-creator',
  'marketing-analyst': '/marketing-analyst',
};

/* Blue Diamond — each role gets a shade within the sapphire crystal palette */
export const ROLE_COLORS: Record<UserRole, string> = {
  'super-admin': 'bg-brand-900/10 text-brand-900 border border-brand-200',
  'marketing-manager': 'bg-brand-100 text-brand-800',
  'content-creator': 'bg-diamond-100 text-diamond-800',
  'marketing-analyst': 'bg-brand-50 text-brand-700 border border-brand-200',
};

export const ROLE_QUICK_ACCESS: Record<UserRole, { button: string; icon: string; card: string; ring: string }> = {
  'super-admin': {
    button: 'btn-role btn-role-deep',
    icon: 'bg-gradient-to-br from-brand-800 to-brand-950 text-white shadow-crystal',
    card: 'border-brand-300/80 hover:border-brand-400 bg-brand-50/70',
    ring: 'ring-brand-700',
  },
  'marketing-manager': {
    button: 'btn-role btn-role-primary',
    icon: 'bg-gradient-to-br from-brand-500 to-brand-700 text-white shadow-crystal',
    card: 'border-brand-200/80 hover:border-brand-400 bg-brand-50/60',
    ring: 'ring-brand-500',
  },
  'content-creator': {
    button: 'btn-role btn-role-crystal',
    icon: 'bg-gradient-to-br from-diamond-400 to-brand-500 text-white shadow-crystal',
    card: 'border-diamond-200/80 hover:border-diamond-400 bg-diamond-50/60',
    ring: 'ring-diamond-500',
  },
  'marketing-analyst': {
    button: 'btn-role btn-role-ice',
    icon: 'bg-gradient-to-br from-diamond-300 to-diamond-500 text-brand-900 shadow-crystal',
    card: 'border-diamond-200/80 hover:border-brand-300 bg-diamond-50/50',
    ring: 'ring-diamond-400',
  },
};

export const ROLE_SIDEBAR_ACCENT: Record<UserRole, { active: string; icon: string; border: string }> = {
  'super-admin': {
    active: 'bg-gradient-to-r from-brand-800 to-brand-950 text-white shadow-md shadow-brand-900/30',
    icon: 'text-brand-800',
    border: 'border-brand-300',
  },
  'marketing-manager': {
    active: 'bg-gradient-to-r from-brand-600 to-brand-800 text-white shadow-md shadow-brand-600/25',
    icon: 'text-brand-600',
    border: 'border-brand-200',
  },
  'content-creator': {
    active: 'bg-gradient-to-r from-diamond-500 to-brand-600 text-white shadow-md shadow-diamond-500/25',
    icon: 'text-diamond-600',
    border: 'border-diamond-200',
  },
  'marketing-analyst': {
    active: 'bg-gradient-to-r from-diamond-400 to-brand-500 text-brand-950 shadow-md shadow-diamond-400/25',
    icon: 'text-diamond-600',
    border: 'border-diamond-200',
  },
};

export const ROLE_ACTION_ACCENT: Record<UserRole, string> = {
  'super-admin': 'bg-gradient-to-br from-brand-100 to-brand-200 text-brand-700',
  'marketing-manager': 'bg-gradient-to-br from-brand-100 to-brand-300 text-brand-700',
  'content-creator': 'bg-gradient-to-br from-diamond-100 to-diamond-200 text-diamond-700',
  'marketing-analyst': 'bg-gradient-to-br from-diamond-100 to-brand-200 text-brand-700',
};
